Readme updates for granular control (#1017)
actions/cache: restore/README.md

## Inputs

* `path` - A list of files, directories, and wildcard patterns to cache and restore. See [`@actions/glob`](https://github.com/actions/toolkit/tree/main/packages/glob) for supported patterns.
* `key` - String used while saving cache for restoring the cache
* `restore-keys` - An ordered list of prefix-matched keys to use for restoring stale cache if no cache hit occurred for key.
> **Note**
It is very important to use the same `key` and `path` that were used by either `actions/cache` or `actions/cache/save` while saving the cache.

### Environment Variables
* `SEGMENT_DOWNLOAD_TIMEOUT_MINS` - Segment download timeout (in minutes, default `60`) to abort download of the segment if not completed in the defined number of minutes. [Read more](https://github.com/actions/cache/blob/main/workarounds.md#cache-segment-restore-timeout)

## Outputs

* `cache-hit` - A boolean value to indicate an exact match was found for the key. 

> * `key` - 

Author: Will add appropriate info here

## Inputs

* `path` - A list of files, directories, and wildcard patterns to cache and restore. See [`@actions/glob`](https://github.com/actions/toolkit/tree/main/packages/glob) for supported patterns.
* `key` - String used while saving cache for restoring the cache
* `restore-keys` - An ordered list of prefix-matched keys to use for restoring stale cache if no cache hit occurred for key.
> **Note**
It is very important to use the same `key` and `path` that were used by either `actions/cache` or `actions/cache/save` while saving the cache.

### Environment Variables
* `SEGMENT_DOWNLOAD_TIMEOUT_MINS` - Segment download timeout (in minutes, default `60`) to abort download of the segment if not completed in the defined number of minutes. [Read more](https://github.com/actions/cache/blob/main/workarounds.md#cache-segment-restore-timeout)

## Outputs

* `cache-hit` - A boolean value to indicate an exact match was found for the key. 
* `cache-primary-key` - Cache primary key passed in the input to use in subsequent steps of the workflow

> * `cache-restore-key` - Cache key restored

Author: Will update this to a new name

# Restore action

> The restore action restores cache just like the main `cache` action, except it doesn't save the cache. So for cases where users only want to restore cache and not save anything in return, this action is pretty helpful. Moreover, this action accepts the same set of inputs as the `cache` action.

Reviewer: ```suggestion
The restore action, as the name suggest, restores a cache. It acts similar to the`cache` action except that it doesn't have a post step to save the cache. This action can provide you a granular control to only restore a cache without having to necessarily save it.  It accepts the same set of inputs as the `cache` action.
```

## Inputs

* `path` - A list of files, directories, and wildcard patterns to cache and restore. See [`@actions/glob`](https://github.com/actions/toolkit/tree/main/packages/glob) for supported patterns.
* `key` - String used while saving cache for restoring the cache
* `restore-keys` - An ordered list of prefix-matched keys to use for restoring stale cache if no cache hit occurred for key.

> > **Note**

Reviewer: Do we need this? It is good for user to use same keys but they can use any key they want.
This statement may scare them.

Author: Can be removed, existing users would be knowing this already. But I just added this for the new users to know that the key+path combination has to be present in both actions to restore/save the same cache. 

We don't want users to file bugs for these trivial cases due to lack of info. 😅

It is very important to use the same `key` and `path` that were used by either `actions/cache` or `actions/cache/save` while saving the cache.

### Environment Variables
* `SEGMENT_DOWNLOAD_TIMEOUT_MINS` - Segment download timeout (in minutes, default `60`) to abort download of the segment if not completed in the defined number of minutes. [Read more](https://github.com/actions/cache/blob/main/workarounds.md#cache-segment-restore-timeout)

> ## Outputs

Reviewer: May be outputs section should come before environment variables one.

Author: We are following the same sequence in the cache action readme as well.

Reviewer: I was thinking from user perspective. Output is more common need than env. In a doc we should start with more commonly useful info at top and then go down to less common useful ones.

Reviewer: I was thinking from user perspective. Output is more common need than env. In a doc we should start with more commonly useful info at top and then go down to less common useful ones.

Author: Updated

    id: cache
    with:
      path: path/to/dependencies
      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
```

#### Step 2 - Restore the built artifact from cache using the same key and path
```
steps:
  - uses: actions/checkout@v3

  - uses: actions/cache/restore@v3
    id: cache
    with:
      path: path/to/dependencies
      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}

  - name: Install Dependencies
    if: steps.cache.outputs.cache-hit != 'true'
    run: /install.sh

  - name: Build
    run: /build-child-module.sh

  - name: Publish package to public
    run: /publish.sh
```

### Exit workflow on cache miss

You can use the output of this action to exit the workflow on cache miss. This way you can restrict your workflow to only initiate the build when `cache-hit` occurs, in other words, cache with exact key is found.

```
steps:
  - uses: actions/checkout@v3

  - uses: actions/cache/restore@v3
    id: cache
    with:
      path: path/to/dependencies
      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}

  - name: Check cache hit
    if: steps.cache.outputs.cache-hit != 'true'
    run: exit 1

  - name: Build
    run: /build.sh
```

## Tips

> Since this action comes with its own set of pros and cons, we are listing some of the ways by which you can tackle the limitations.

Reviewer: Don't call them limitations. This whole statement is not useful
```suggestion
```

### Reusing primary key and restored key in the save action

> One of the limitation you might experience is passing the same input in both `actions/cache/restore` and `actions/cache/save` action. To avoid this, you can make use of the `outputs` from the restore action to reuse the same primary key and also the key of the cache that was restored. This way changing any key in the restore action will automatically reflect in the subsequent actions where the same input is being used.

Reviewer: ```suggestion
Usually you may want to use same `key` in both actions/cache/restore` and `actions/cache/save` action. To achieve this, use `outputs` from the restore action to reuse the same primary key (or the key of the cache that was restored).
```
